pep-0696.rst

```diff
@@ -160,19 +161,21 @@ literal "``...``".
 
    DefaultTs = TypeVarTuple("DefaultTs", default=Unpack[tuple[str, int]])
 
-   class Foo(Generic[DefaultTs]): ...
+   class Foo(Generic[*DefaultTs]): ...
 
    reveal_type(Foo())             # type is Foo[str, int]
    reveal_type(Foo[int, bool]())  # type is Foo[int, bool]
 
-Using another ``TypeVarLike`` as the default
+Using Another ``TypeVarLike`` as ``default``
```

Comment:
Ok so coming back to this, I'm pretty confident that this is possible to implement (this sure isn't perfect and I had to disable some mypy sanity checks for things like default=T where T didn't use to be valid being passed) https://github.com/python/mypy/compare/master...Gobot1234:mypy:TypeVar-defaults.

```py
from typing import Callable, Generic, TypeVar
from typing_extensions import ParamSpec, TypeAlias, TypeVarTuple, Unpack, reveal_type

T = TypeVar("T", default=int)
T2 = TypeVar("T2", default=Callable[[T], int])


class Foo(Generic[T, T2]): ...


reveal_type(Foo)  # revealed type is type[__main__.Foo[T = builtins.int, T2 = (T = builtins.int) -> builtins.int]]
reveal_type(Foo[str])  # revealed type is type[__main__.Foo[builtins.str, T2 = (builtins.str) -> builtins.int]]
reveal_type(Foo[str, int])  # revealed type is type[__main__.Foo[builtins.str, int]]

PreSpecialised: TypeAlias = Foo[str]
reveal_type(PreSpecialised)    # revealed type is type[__main__.Foo[builtins.str, T2 = (builtins.str) -> builtins.int]]
reveal_type(PreSpecialised[int])  # This currently doesn't work but I don't think it will be hard to fix


P = ParamSpec("P", default=[int, str])
P2 = ParamSpec("P2", default=P)

class Bar(Generic[P, P2]): ...
def foo(fn: Callable[P, int]) -> bool: ...
reveal_type(foo)  # revealed type is def [P = [builtins.int, builtins.str]] (fn: (*P.args, **P.kwargs) -> builtins.int) -> builtins.bool
reveal_type(Bar)  # revealed type is type[__main__.Bar[P = [builtins.int, builtins.str], P2 = P = [builtins.int, builtins.str]]]
```
(I've slightly altered the representations mypy gives to make things a tiny bit clearer)
FYI this doesn't support TypeVarTuple defaults just cause mypy doesn't support PEP 646 yet, but enabling them should be about 20 to 30 lines of code

Reply:
I've just found out mypy does sort of support TypeVarTuples so I'll add support for that

Reply:
Ok this is actually a lot harder to support than I initially thought, I think I'm best waiting before trying to add this to mypy, there are a lot of gaps in the TypeVarTuple support. I can PR this feature in further down the road.